avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -206,10 +221,14 @@ public static boolean isSupportedForFastSerializer(Schema.Type schemaType) {
           new FastDeserializerWithAvroSpecificImpl<>(writerSchema, readerSchema));
       if (deserializer == null) {
         deserializer = fastSpecificRecordDeserializersCache.get(schemaKey);
-        CompletableFuture.supplyAsync(() -> buildSpecificDeserializer(writerSchema, readerSchema), executor)
-            .thenAccept(d -> {
-              fastSpecificRecordDeserializersCache.put(schemaKey, d);
-            });
+        if (schemaNum.get() < schemaNumLimit) {
+          CompletableFuture.supplyAsync(() -> buildSpecificDeserializer(writerSchema, readerSchema), executor).thenAccept(d -> {
+            fastSpecificRecordDeserializersCache.put(schemaKey, d);
+          });
+        } else if (schemaNum.get() == schemaNumLimit) {
+          LOGGER.info("Generated fast de/serializer number hits " + schemaNumLimit + " limit");
```

Comment:
make this a warning. this is definitely something oenwers of running code should pay attention to

Reply:
also, can you print the value of the counter as well as the max? without it users wont have a clue what they should set the limit to